internal/provider/data_file_test.go

```diff
@@ -0,0 +1,107 @@
+package provider_test
+
+import (
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/neuspaces/terraform-provider-system/internal/acctest"
+	"github.com/neuspaces/terraform-provider-system/internal/acctest/tfbuild"
+	"testing"
+)
+
+func TestAccDataFile_read_content(t *testing.T) {
+	testConfig := newTestFileConfig()
+
+	acctest.Current().Targets.Foreach(t, func(t *testing.T, target acctest.Target) {
+		t.Parallel()
+
+		resource.Test(t, resource.TestCase{
+			ProviderFactories: acctest.ProviderFactories(),
+			Steps: []resource.TestStep{
+				{
+					Config: tfbuild.FileString(tfbuild.File(
+						acctest.ProviderConfigBlock(target.Configs.Default()),
+						testAccFileBlock("test", testRunFilePath(target, testConfig.fileName),
+							tfbuild.AttributeString("mode", "644"),
+							tfbuild.AttributeString("content", "hello world!"),
+						),
+						testAccDataFileBlock("test", testRunFilePath(target, testConfig.fileName)),
+					)),
+					Check: resource.ComposeTestCheckFunc(
+						resource.TestCheckResourceAttr("system_file.test", "id", testRunFilePath(target, testConfig.fileName)),
```

Comment:
Please remove any `resource.TestCheckResourceAttr` which is related to `system_file.test` because the resource is not subject to this acceptance test.